app/src/main/java/com/kamron/pogoiv/OcrHelper.java

```diff
@@ -153,6 +153,92 @@ private double getPokemonLevelFromImg(Bitmap pokemonImage, int trainerLevel) {
+        if (stringCacheEvoCandyCost != null) {
+            return Integer.parseInt(stringCacheEvoCandyCost);
+        }
+
+        //clean the image
+        //the dark color used for text in pogo is approximately rgb 76,112,114 if you can afford evo
+        //and the red color is rgb 255 95 100 when you cant afford the evolution
+        Bitmap evolutionCostImageCanAfford = replaceColors(evolutionCostImage, false, 80, 110, 110, Color.WHITE, 45,
+                false);
+        Bitmap evolutionCostImageCannotAfford = replaceColors(evolutionCostImage, false, 255, 95, 100, Color.WHITE, 30,
+                false);
+
+        boolean affordIsBlank = isOnlyWhite(evolutionCostImageCanAfford);
+        boolean cannotAffordIsBlank = isOnlyWhite(evolutionCostImageCannotAfford);
+        //check if fully evolved
+        if (affordIsBlank && cannotAffordIsBlank) { //if there's no red or black text, there's no text at all.
+            ocrCache.put(hash, "-1");
+            return -1;
+        }
+
+        //use the correctly refined image (refined for red or black text)
+        if (affordIsBlank) {
+            evolutionCostImage = evolutionCostImageCannotAfford;
+        } else {
+            evolutionCostImage = evolutionCostImageCanAfford;
+        }
+
+        //If not cached or fully evolved, ocr text
+        int result;
+        tesseract.setImage(evolutionCostImage);
+        String ocrResult = fixOcrLettersToNums(tesseract.getUTF8Text());
+        ocrResult = ocrResult.replaceAll("[^\\d]", ""); //remove all non numbers
+        try {
+            result = Integer.parseInt(ocrResult);
+            if(result == 10){ //second zero hidden behind floating button
+                result = 100 ;
+                ocrResult = "100";
+            }
+            ocrCache.put(hash, ocrResult);
+        } catch (NumberFormatException e) {
+            result = error; //could not ocr text
+        }
+        return result;
+
+    }
+
+    /**
+     * Heuristic method to determine if the image looks empty. Works by taking a horisontal row of pixels from he
+     * middle, and looks if they're all pure white.
+     *
+     * @param refinedImage A pre-processed image of the evolution cost. (should be pre-refined to replace all non
+     *                     text colors with pure white)
+     * @return true if the image is likely only white
+     */
+    private boolean isOnlyWhite(Bitmap refinedImage) {
+        int[] pixelArray = new int[refinedImage.getWidth()];
+
+        //below code takes one line of pixels in the middle of the pixture from left to right
+        refinedImage.getPixels(pixelArray, 0, refinedImage.getWidth(), 0, refinedImage.getHeight() / 2, refinedImage
+                .getWidth(), 1);
+
+        // a loop that sums the color values of all the pixels in the array
+        for (int pixel : pixelArray) {
+            if (Color.red(pixel) != 255 || Color.green(pixel) != 255 || Color.blue(pixel) != 255) {
```

Comment:
I think this condition could be replaced by `pixel != Color.rgb(255, 255, 255)`, like we do elsewhere; that should be faster.